packages/react-ui/components/CurrencyLabel/__tests__/CurrencyLabel.test.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+
+import { THIN_SPACE } from '../../../components/CurrencyInput/CurrencyHelper';
+import { CurrencyLabel, CurrencyLabelDataTids } from '../CurrencyLabel';
+
+const SPACE_SYMBOL = String.fromCharCode(THIN_SPACE);
+
+describe('CurrencyLabel', () => {
+  it('should correctly format value', () => {
+    render(<CurrencyLabel value={12345} fractionDigits={0} />);
+
+    const node = screen.getByTestId(CurrencyLabelDataTids.root);
+    expect(node).toHaveTextContent('12 345');
+  });
+
+  it('should render trailing zeros', () => {
+    render(<CurrencyLabel value={12356.14} fractionDigits={5} />);
+
+    const node = screen.getByTestId(CurrencyLabelDataTids.root);
+
+    // .toHaveTextContent normalizes input value in the way that makes test always pass: https://github.com/testing-library/jest-dom/blob/948d90f32cc79339bdeebea0454599db74c5d071/src/to-have-text-content.js#L10-L12
+    // eslint-disable-next-line jest-dom/prefer-to-have-text-content
+    expect(node.textContent).toBe(`12${SPACE_SYMBOL}356,14000`);
```

Comment:
Кажется, нормализация пробелов в этом случае нам только в помощь.
Предлагаю не париться, и использовать обычный пробел в тестах.

При желании, наличие правильного пробела можно проверять отдельным тестом.

Reply:
В документации метода [toHaveTextContent](https://github.com/testing-library/jest-dom#tohavetextcontent) указано, что при передаче в метод строки - выполняется лишь частичное сравнение с контентом:
> _When a `string` argument is passed through, it will perform a **partial ... match** to the node content._

А для сравнения всего контента рекомендуется использовать регулярку:
> _If you want to **match the whole content**, you can use a `RegExp` to do it._

Т.е. в текущем тест-кейсе можно сделать так:
```
expect(node).toHaveTextContent(/^12 356,14000$/)
```

Reply:
Переписал на регулярку во всех местах где используется `.toHaveTextContent`, так как `.textContent` + `.toBe` по умолчанию мэтчат строку полностью